extensions/blocks/dialogue/components/participants-control.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	TextControl,
+	BaseControl,
+	SelectControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+
+function ParticipantsMenu( { participants, className, onSelect } ) {
+	return (
+		<MenuGroup className={ `${ className }__participants-selector` }>
+			{ map( participants, ( { participant, participantSlug } ) => (
+				<MenuItem
+					key={ participantSlug }
+					onClick={ () => onSelect( { participantSlug } ) }
+				>
+					{ participant }
+				</MenuItem>
+			) ) }
+		</MenuGroup>
+	);
+}
+
+function ParticipantControl( { className, participantValue, onChange } ) {
+	return (
+		<BaseControl className={ `${ className }__custom-participant` }>
+			<div className={ `${ className }__text-button-container` }>
+				<TextControl
+					label={ __( 'Custom', 'jetpack' ) }
+					value={ participantValue }
+					onChange={ ( participant ) => onChange( {
+						participantSlug: null,
+						participant,
+					} ) }
+					onFocus={ ( { target } ) => onChange( {
+						participantSlug: null,
+						participant: target?.value,
+					} ) }
+				/>
+			</div>
+		</BaseControl>
+	);
+}
+
+export function ParticipantsControl( {
+	participants,
+	currentParticipantSlug,
+	onSelect,
+} ) {
+		return (
+			<SelectControl
+				label={ __( 'Participant name', 'jetpack' ) }
+				value={ currentParticipantSlug }
```

Comment:
I'm seeing a warning about this being null (at least I think this is the code that's generating the warning)

<img width="812" alt="Screen Shot 2020-12-07 at 21 08 05" src="https://user-images.githubusercontent.com/1699996/101434698-a9ea7500-38d0-11eb-858e-8361ed8817e8.png">

Reply:
It's weird I don't see these warning :-/

Reply:
Although I wasn't getting the warning, which is weird, I've added an empty string as default. Could you confirm it doesn't show the warning after this change? 5c541a6f4811528d47bdc3c89bb488751ff623b4 Feel free to mark it as resolved.

Reply:
Confirmed that the `value` prop warning is fixed now